Add REF view to references method

fabiosangregorio opened the pull request:
I added the REF view to the references method in AbstractRetrieval. REF view introduces more authors info, so I introduced a namedtuples for the authors.
REF also includes the Scopus EID and the citedbycount fields, which I added to the existing Reference namedtuples.
I updated the references test case to include the newly added fields.

Fixes #81

Michael-E-Rose commented:
This is look amazing, and it's also great that you updated the tests.  But I wonder whether creating a namedtuple as a field is the best way.  After all, it's a namedtuple in a nameduple.  Turning the outer namedtuple into a DataFrame will leave another namedtuple in a cell.

What speaks against having separate fields for the author information?

Another question: Why adding `scopuseid` if the `id` (i.e. the Scopus ID) is already present?

fabiosangregorio commented:
I thought grouping the author's information into a single object would be better, so that you can have all the author's info in one place. Also it doesn't break the current FULL view structure (you still have a list of authors with all their info in it). 
But as you said, it's a namedtuple in a namedtuple. Do you think it's better to have separate fields for the author? How would you do it?

Regarding the scopus EID, I'm kinda new to Scopus so I don't know much about the details of the IDs. I just added it because it's a field available in the REF view. If it's unnecessary we can remove it.

Michael-E-Rose commented:
Yes, EID as a separate field is definitely not relevant, as "2-s2.0-" plus Scopus ID makes the EID.  The 2 in the beginning marks it as Document.  Authors for example have a 9.

I will think a bit a day about the fields.  The alternative would be to have new fields joining the relevant information: "scopus_id1;scopus_id2;scopus_id3", etc.  That's at least the approach of other fields that return otherwise nested information as namedtuples.

fabiosangregorio commented:
Ok, we can omit it then. Thank you for the explanation.
In my opinion, having the authors as namedtuples is not a big deal, and since the API returns quite a lot of information about them, I think it would be the clearest approach.

In any case, let me know how we should implement this.

Michael-E-Rose commented:
Hi! Yes, I have been thinking about it and asked my colleagues (which gives some estimate of how the class will be used). The three most common use cases are:
1. Building an author network based on citations
2. Checking for self-cites
3. Looking into search results in a csv file

For the first, I think both namedtuples and separate fields are equally easy with itertools' product and require three lines of code equally (two namedtuples probably require
````

fabiosangregorio commented:
So the reference namedtuple would look something like this?
```python
ref = namedtuple('Reference', 'position id doi title authors_id authors_name authors_seq '\
            'authors_affiliationid authors_affiliationhref authors_auid authors_url sourcetitle '\
            'publicationyear volume issue first last text fulltext scopuseid citedbycount')
```

in which every authors_field is the field of every author joined together, just like this, right?
```python
authors_url = 'auth1_url;auth2_url;auth3_url'
```

Thank looks a bit counterintuitive at first, but if this is needed in order to be able to export the results to csv I'm okay with it. 
Shall I implement this?

Michael-E-Rose commented:
Yes, that looks right and it would be amazing if you would implement this!

fabiosangregorio commented:
Hi, I splitted the authors fields in separate fields.

Instead of joining all the auhtors_field together, I put them in separate arrays, just like the authors' names in the current version of your repo.
I also removed the `@seq` field as it contains the author's position in the reference, and it can be obtained using the author's index in the list.

For backward compatibility, the authors' names are contained in the `authors` list, just like the current version of your repo.
I also removed the scopus EID and updated the references test.

Can this work?

Michael-E-Rose commented:
Thank you Fabio for the good work!  I tested it any it seems to work.  Would you like to add one more test case to `test_AbstractRetrieval.py` with one abstract (of your choice) which would be loaded with `view="REF"`?  This would make the entire PR be completely self-contained :)

fabiosangregorio commented:
Looking at the huge test case, I realize that `authors_url` and `authors_affiliationurl` are redundant information, since they're obtainable from the ids. Therefore I am going to remove them.

fabiosangregorio commented:
@Michael-E-Rose this should be it :) Sorry about the confusion, it's my first PR on Github 🎉
